lib/ansiblelint/errors.py

```diff
@@ -27,3 +34,19 @@ def __repr__(self):
 
         return formatstr.format(_id, self.message,
                                 self.filename, self.linenumber, self.line)
+
+    @property
+    def _hash_key(self):
+        return self.filename, self.linenumber, self.message, getattr(self.rule, 'id', 0)
```

Comment:
Side note: should we cache this?

Reply:
No. If we do, it will count as premature optimization. Lets keep optimizations where it really affects total execution time.